### PR TITLE
SCMOD-7839: Added batch file clean up 

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**

### CODE REVIEW

**Comment by Anthony-McGreevy** [comment]
@andyreidz I am still testing this but wanted to see if you would mind reviewing the logic please.

File: staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public static String getSubBatchName()
+    {
+        return getCurrentTimeAsString().concat(SUBBATCH_FILE_SUFFIX);
+    }
+
+    private static String getCurrentTimeAsString()
+    {
+        return cleanseTime(Instant.now().toString());
```
Comment (andyreidz) [inline staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java:42]: Does toString produce UTC?

File: staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public static long getFileCreationTime(final String fileName)
+    {
+        final String time = fileName.substring(0, 22);
+
+        try {
+            System.out.println("File creation time file name: " + time);
+            return Instant.parse(reverseCleanseTimeString(time)).toEpochMilli();
```
Comment (andyreidz) [inline staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java:51]: What timezone will this use?

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/InprogressBatchCleanUpController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging;
+
+import com.github.cafdataprocessing.services.staging.dao.BatchDao;
+import org.springframework.stereotype.Component;
+
+@Component("batchCleanUpController")
+public class InprogressBatchCleanUpController
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/InprogressBatchCleanUpController.java]: Your class name suggests this class is a @Controller

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/InprogressBatchCleanUpController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging;
+
+import com.github.cafdataprocessing.services.staging.dao.BatchDao;
+import org.springframework.stereotype.Component;
+
+@Component("batchCleanUpController")
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/InprogressBatchCleanUpController.java]: I don't think "batchCleanUpController" is needed.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/InprogressBatchCleanUpController.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+public class InprogressBatchCleanUpController
+{
+    private final BatchDao fileSystemDao;
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/InprogressBatchCleanUpController.java]: it's a BatchDao not a fileSystemDao

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java [hunk 6]
```diff
@@ -234,4 +244,91 @@ private void completeInProgressBatch(final TenantId tenantId, final Path inProgr
...
+
+    /**
+     * Returns a list of the path to all sub files in all folders provide
+     * @param tenantFolders List of Path objects that represent all current tenant in_progress folders
+     * @return return a Stream of paths to all subfiles under all folders provided
+     */
+    private Stream<Path> getAllBatchFilesForAllDirectories(final Path tenantFolder)
+    {
+        try {
+            return Files.list(tenantFolder);
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java:322]: * <p> The returned stream encapsulates one or more {@link DirectoryStream}s.
     * If timely disposal of file system resources is required, the
     * {@code try}-with-resources construct should be used to ensure that the

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java [hunk 7]
```diff
@@ -234,4 +244,91 @@ private void completeInProgressBatch(final TenantId tenantId, final Path inProgr
...
+            }
+        } catch (final IOException ex) {
+            LOGGER.error("An exception occured trying to read the files in the base directory.", ex);
+        }
+    }
+
+    private boolean checkAllSubfilesSafely(final Path path)
+    {
+        try {
+            return Files.list(path)
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java:286]: * <p> The returned stream encapsulates one or more {@link DirectoryStream}s.
     * If timely disposal of file system resources is required, the
     * {@code try}-with-resources construct should be used to ensure that the

**Review by andyreidz — CHANGES_REQUESTED** [review]
InprogressBatchCleanUpController feels a bit pointless, since all the logic is in the BatchDao why not use that from 
    @Scheduled(fixedDelayString = "${staging.fileCleanUpInterval}")
    public void cleanUpAbandonedBatches()
    {

File: staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java (hunk 1 above)
Comment (Anthony-McGreevy) [reply staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java:42]: Yes

File: staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java (hunk 2 above)
Comment (Anthony-McGreevy) [reply staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java:51]: UTC, that is why we have to do the reverse cleanse, to have the UTC time in the file name we need to remove the colons from the time. To parse it back we have to put them back in before parsing.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java (hunk 7 above)
Comment (Anthony-McGreevy) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java:286]: If I do that won't the steam be closed once this method exits? I want the stream returned to the caller to operate on.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java (hunk 6 above)
Comment (Anthony-McGreevy) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java:322]: If I do that won't the steam be closed once this method exits? I want the stream returned to the caller to operate on.

**Review by andyreidz — APPROVED** [review]

**Pull request merged**